modules/microvm/virtualization/microvm/guivm.nix

```diff
@@ -230,9 +236,21 @@ in
     microvm.vms."${vmName}" = {
       autostart = true;
       config = guivmBaseConfiguration // {
-        boot.kernelPackages = lib.mkIf config.ghaf.guest.kernel.hardening.graphics.enable (
-          pkgs.linuxPackagesFor guest_graphics_hardened_kernel
-        );
+        boot.kernelPackages =
+          if config.ghaf.guest.kernel.hardening.graphics.enable then
+            pkgs.linuxPackagesFor guest_graphics_hardened_kernel
+          else
+            # Align Kernel version which is used by ghaf-host
+            config.boot.zfs.package.latestCompatibleLinuxPackages;
```

Comment:
does the gui-vm need the zfs compatible kernel?

Reply:
Not necessary but will `pkgs.linuxPackages_latest` (6.10.2) default brightness value of ghaf system will change from `96000`(max brightness) to `4800` (5% of max brightness).
There could be some changes in backlight driver.
Verified using `cat /sys/class/backlight/intel_backlight/brightness`